06.bowling_object/game.rb

```diff
@@ -0,0 +1,47 @@
+# frozen_string_literal: true
+
+require_relative 'frame'
+
+class Game
+  FINAL_FRAME_SHOT = 18
+  FINAL_FRAME = 9
+
+  def initialize(result)
+    @result = result
+  end
+
+  def calculate_total_score
+    frames = generate_frames(@result)
```

Comment:
`@result`はインスタンス変数でクラス内から自由にアクセス可能なので、引数で渡さなくても使えます。

```suggestion
  def calculate_total_score
    frames = generate_frames
```

ただ、引数で渡した方がコードの見通しがよくなるというメリットもあります。
つまりどっちでもいいということです。
これはもうちょっと突き詰めると、インスタンス変数であってもなくてもどっちでもいい、ということです。
よって、次のようなメソッド設計にもできます。

```suggestion
  def calculate_total_score(result)
    frames = generate_frames(result)
```

インスタンス変数はスコープが広い（寿命が長い）変数なので、なるべく減らした方がよいです。
なので`@result`はなくした方がいいかもしれませんね。

Reply:
`@result`はなくし、`calculate_total_score`メソッドの引数に`result`を渡しました。